src/NHibernate/Criterion/CriterionUtil.cs

```diff
@@ -46,6 +46,11 @@ public static class CriterionUtil
 
 		internal static SqlString[] GetColumnNamesUsingProjection(IProjection projection, ICriteriaQuery criteriaQuery, ICriteria criteria)
 		{
+			if (projection is IPropertyProjection propertyProjection)
```

Comment:
This change is actually unrelated to fix. But it fixes cases when `GetColumnNamesUsingProjection` is called for composite property projections (in case of any in the future)